src/gui.ml

```diff
@@ -154,11 +155,51 @@ let draw_position (bd : Game.t) (imgs : image list) =
   let bd = board_to_array bd in
   draw_position_rows bd imgs 0
 
-let draw_game (bd : Game.t) =
+let rec get_potential_squares (move_list : move list) currX currY :
+    (int * int) list =
+  match move_list with
+  | ((a, b), (c, d)) :: t when currX = a && currY = b ->
+      (c, d) :: get_potential_squares t currX currY
+  | h :: t -> get_potential_squares t currX currY
+  | _ -> []
+
+let draw_potential_circles
+    (move_list : move list)
+    (imgs : image list)
+    currX
+    currY =
+  let potential_moves = get_potential_squares move_list currX currY in
+  let rec draw_circle (potential_moves : (int * int) list) =
+    match potential_moves with
+    | (a, b) :: t ->
+        draw_image (List.nth imgs 12)
+          ((a * step) + ((step - 60) / 2))
+          ((b * step) + ((step - 60) / 2));
+        draw_circle t
+    | [] -> ()
+  in
+  draw_circle potential_moves
+
+(** [draw circles bd imgs] draws grey circles based on the potential
+    moves of the clicked square. *)
+let draw_circles
+    (bd : Game.t)
+    (imgs : image list)
+    currX
+    currY
```

Comment:
Whenever you have this, just make it a single tuple input. You can pattern match it, but I think we should keep it consistent and use (currX, currY) instead of two separate arguments.